src/Lykke.Sdk/SdkModule.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using Autofac;
+using AzureStorage.Tables;
+using Common.Log;
+using Lykke.Logs;
+using Lykke.Sdk.Settings;
+using Lykke.SettingsReader;
+using Lykke.SlackNotification.AzureQueue;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+
+namespace Lykke.Sdk
+{
+    internal class SdkModule<TAppSettings> : Module 
+        where TAppSettings : BaseAppSettings
+    {
+        private readonly Func<IReloadingManager<TAppSettings>, IReloadingManager<string>> _logsConnectionStringFactory;
+        
+        public SdkModule(Func<IReloadingManager<TAppSettings>,IReloadingManager<string>> logsConnectionStringFactory)
+        {
+            _logsConnectionStringFactory = logsConnectionStringFactory;
+        }
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder.Register(ctx => new ConfigurationBuilder().AddEnvironmentVariables().Build())
+                .As<IConfigurationRoot>().SingleInstance();
+
+            builder.Register(ctx => ctx.Resolve<IConfigurationRoot>().LoadSettings<TAppSettings>())
```

Comment:
I think will be better to load setting explicitly in the ```BuildServiceProvider``` because, if we'll load settings here, in the ```Register``` lambda, then in case of incorrect settings, exception will contain to many excess autofac-specific information. Devops will be not happy to read all this autofac stuff